### PR TITLE
Display book details

### LINKED ISSUE #14: Display book details
Click on a book to see overlay with book details

### PR DESCRIPTION
**Pull request by skomputer**
Closes #14.

### CODE REVIEW

File: src/OPDSDataAdapter.ts [hunk 1]
```diff
@@ -25,13 +26,26 @@ function entryToBook(entry: any, feedUrl: string): BookProps {
...
   }
+
+  let detailUrl;
+  let detailLink = entry.links.find(link => link instanceof CompleteEntryLink);
+  if (detailLink) {
+    detailUrl = detailLink.href;
+  }
+
+  // until OPDSParser parses dcterms:publisher...
+  let publisher = entry.unparsed && entry.unparsed["dcterms:publisher"] ? entry.unparsed["dcterms:publisher"][0]["_"] : null;
```
Comment (aslagle) [inline src/OPDSDataAdapter.ts]: this won't work if someone uses a different namespace for dcterms, but we can fix that when we move it to the parser

File: src/OPDSDataAdapter.ts [hunk 2]
```diff
@@ -70,6 +102,7 @@ export function feedToCollection(feed: any, feedUrl: string): CollectionProps {
   let laneIndex = [];
   let facetGroups: FacetGroupProps[] = [];
   let search: SearchProps;
+  let selfUrl: string;
```
Comment (aslagle) [inline src/OPDSDataAdapter.ts]: this isn't getting set anywhere

File: src/__tests__/OPDSDataAdaptor-test.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ describe("OPDSDataAdapter", () => {
     expect(book.authors.length).toEqual(2);
     expect(book.authors[0]).toEqual(entry.authors[0].name);
     expect(book.authors[1]).toEqual(entry.authors[1].name);
-    expect(book.summary).toEqual(entry.summary);
+    expect(book.summary).toEqual(entry.summary.content);
```
Comment (aslagle) [inline src/__tests__/OPDSDataAdaptor-test.ts]: check publisher and published too

File: src/components/Book.tsx [hunk 4]
```diff
@@ -34,7 +35,13 @@ export default class Book extends React.Component<BookProps, any> {
...
       <div className="book" style={ bookStyle }>
-        <img src={this.props.imageUrl} style={ bookCoverStyle } />
+        <BookPreviewLink
+          url={this.props.url}
+          collectionUrl={this.props.collectionUrl}
+          book={this.props}
+          showBookDetails={this.props.showBookDetails}>
+          <img src={this.props.imageUrl} style={bookCoverStyle} />
+        </BookPreviewLink>
         <div className="bookInfo" style={ bookInfoStyle }>
```
Comment (aslagle) [inline src/components/Book.tsx]: why wouldn't clicking on the title and author also show the book details?

File: src/components/BookDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            <div style={{ marginTop: "2em", color: "#888", fontSize: "0.9em" }}>
+              <div className="bookDetailsPublished">Published: {this.props.published}</div>
+              <div className="bookDetailsPublisher">Publisher: {this.props.publisher}</div>
+            </div>
+          </div>
+          <div style={{ clear: "both" }}></div>
+          <div
+            className="bookDetailsSummary"
+            style={{ marginTop: "1em", marginBottom: "1em", paddingTop: "1em", borderTop: "1px solid #ccc" }}
+            dangerouslySetInnerHTML={{ __html: this.props.summary }}></div>
```
Comment (aslagle) [inline src/components/BookDetails.tsx]: we'll need to sanitize the summaries

File: src/components/BookPreviewLink.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+export default class BookPreviewLink extends Link<BookLinkProps> {
+  processClick() {
+    this.props.showBookDetails(this.props.book);
+  }
+
+  href() {
+    if (this.props.collectionUrl) {
+      return `?url=${this.props.collectionUrl}&book=${this.props.url}`;
+    } else {
+      return `?book=${this.props.url}`;
```
Comment (aslagle) [inline src/components/BookPreviewLink.tsx]: when does this case happen?

File: src/components/__tests__/Link-test.tsx [hunk 7]
```diff
@@ -1,25 +0,0 @@
...
-
-import Link from "../Link";
-
-let link = {
-  id: "root.xml",
-  title: "First Acquisition feed",
-  href: "acquisition/main.xml"
-};
-
-describe("Link", () => {
```
Comment (aslagle) [inline src/components/__tests__/Link-test.tsx]: I feel like Link should still have some tests

File: src/interfaces.d.ts [hunk 8]
```diff
@@ -85,14 +91,30 @@ interface UrlFormProps extends FetchCollectionProps {
...
+interface BookLinkProps extends LinkProps, BookActionProps {
+  book?: BookProps;
 }
 
 interface ErrorMessageProps {
   message: string;
   closeError: () => void;
+}
+
+interface BaseProps extends __React.HTMLProps<any> {
```
Comment (aslagle) [inline src/interfaces.d.ts]: this interface is also at the top of the file

File: src/components/BookPreviewLink.tsx (hunk 6 above)
Comment (skomputer) [inline src/components/BookPreviewLink.tsx]: permalinks, not currently implemented

Comment (aslagle) [inline src/components/BookPreviewLink.tsx]: but how would there be book preview links when there's no collection?

**Pull request merged**